### PR TITLE
ALTER TABLE MODIFY QUERY support for WindowView

### PR DESCRIPTION
**Pull request by Vxider**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
`ALTER TABLE ... MODIFY QUERY` support for WindowView.

Note that `alter table` on WindowView will lose data in the inner table because the mergeable state cannot reused after the query was changed.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by Vxider** [comment]
@Mergifyio update

**Comment by Vxider** [comment]
@Mergifyio update

**Comment by Vxider** [comment]
@Mergifyio update

File: src/Storages/WindowView/StorageWindowView.cpp [hunk 1]
```diff
@@ -447,6 +449,65 @@ bool StorageWindowView::optimize(
...
+    StorageInMemoryMetadata new_metadata = getInMemoryMetadata();
+    StorageInMemoryMetadata old_metadata = getInMemoryMetadata();
+    params.apply(new_metadata, local_context);
+
+    const auto & new_select = new_metadata.select;
+    const auto & new_select_query = new_metadata.select.inner_query;
+
+    auto old_inner_table_id = inner_table_id;
+
+    modifying_query = true;
```
Comment (kssenii) [inline src/Storages/WindowView/StorageWindowView.cpp:467]: may be better like this?
```suggestion
    modifying_query = true;
    SCOPE_EXIT({
        modifying_query = false;
    });
```

**Review by kssenii — APPROVED** [review]

**Comment by kssenii** [comment]
@Vxider let's resolve conflicts

**Pull request merged**